File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        ref = {}
        ref[0] = 1
        total = 0
        count = 0
        for i in nums:
            total += i
            if total-k in ref:
                count += ref[total - k]
            ref[total] = ref.get(total, 0) + 1
        return count

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> map;
        map[0] = 1;
        int count = 0;
        int sum = 0;
        for(int num : nums) {
            sum += num;
            count += map[sum - k];
            map[sum] = map[sum] + 1;
        }
        return count;
    }
};